circus-train-aws-sns/src/main/java/com/hotels/bdp/circustrain/aws/sns/event/SnsConfiguration.java

```diff
@@ -32,21 +33,30 @@
 import com.amazonaws.auth.AWSCredentialsProviderChain;
 import com.amazonaws.auth.BasicAWSCredentials;
 import com.amazonaws.auth.InstanceProfileCredentialsProvider;
-import com.amazonaws.regions.Region;
-import com.amazonaws.regions.Regions;
+import com.amazonaws.client.builder.ExecutorFactory;
 import com.amazonaws.retry.PredefinedRetryPolicies;
+import com.amazonaws.services.sns.AmazonSNSAsync;
 import com.amazonaws.services.sns.AmazonSNSAsyncClient;
 
 @Configuration
 public class SnsConfiguration {
 
   @Bean
-  AmazonSNSAsyncClient amazonSNS(ListenerConfig config, AWSCredentialsProvider awsCredentialsProvider) {
-    AmazonSNSAsyncClient client = new AmazonSNSAsyncClient(awsCredentialsProvider,
-        new ClientConfiguration().withRetryPolicy(PredefinedRetryPolicies.getDefaultRetryPolicy()),
-        Executors.newSingleThreadScheduledExecutor());
-    client.setRegion(Region.getRegion(Regions.fromName(config.getRegion())));
-    return client;
+  AmazonSNSAsync amazonSNS(ListenerConfig config, AWSCredentialsProvider awsCredentialsProvider) {
+    ClientConfiguration clientConfiguration = new ClientConfiguration().withRetryPolicy(
+        PredefinedRetryPolicies.getDefaultRetryPolicy());
+    ExecutorFactory executorFactory = new ExecutorFactory() {
```

Comment:
what's the default executorFactory, perhaps we don't need to override this at all

Reply:
Same for the retry policy.

Reply:
Thanks for the comments.

If not provided, the retry will indeed default to the same configuration that we are providing. So I will remove this.

The executory will default to `newFixedThreadPool(maxConnections)` where `maxConnections` defaults to 50, so this is slightly different to `newSingleThreadScheduledExecutor()`. Shall I stick with our implementation or go with the default?

Reply:
Hmmmm, I see why we would want to leave it with a single threaded one as that won't change any behaviour. On the other hand unless there's a really good reason to deviate from the defaults we shouldn't (convention over configuration). It's unclear here why we would deviate. Thoughts from anyone else?

Reply:
I'm ok either setting, I suspect it was done to prevent waste. We are just sending a handful of sns messages. Premature optimisation maybe I have no idea.

Reply:
If we're happy to go with the default, I'll go with that instead to stick with convention, thanks both!